fvm/environment/minimum_required_version.go

```diff
@@ -0,0 +1,75 @@
+package environment
+
+import (
+	"github.com/coreos/go-semver/semver"
+
+	"github.com/onflow/flow-go/fvm/storage/state"
+)
+
+// MinimumRequiredVersion returns the minimum required cadence version for the current environment
+// in semver format.
+type MinimumRequiredVersion interface {
+	MinimumRequiredVersion() (string, error)
+}
+
+type minimumRequiredVersion struct {
+	txnPreparer state.NestedTransactionPreparer
+}
+
+func NewMinimumRequiredVersion(
+	txnPreparer state.NestedTransactionPreparer,
+) MinimumRequiredVersion {
+	return minimumRequiredVersion{
+		txnPreparer: txnPreparer,
+	}
+}
+
+func (c minimumRequiredVersion) MinimumRequiredVersion() (string, error) {
+	executionParameters := c.txnPreparer.ExecutionParameters()
+
+	cadenceVersion := mapToCadenceVersion(executionParameters.ExecutionVersion, fvmToCadenceVersionMapping)
+
+	return cadenceVersion.String(), nil
+}
+
+func mapToCadenceVersion(flowGoVersion semver.Version, versionMapping []VersionMapEntry) semver.Version {
+	// return 0.0.0 if there is no mapping for the version
+	var cadenceVersion = semver.Version{}
+
+	greaterThanOrEqualTo := func(version semver.Version, versionToCompare semver.Version) bool {
+		return version.Compare(versionToCompare) >= 0
+	}
+
+	for _, entry := range versionMapping {
+		if greaterThanOrEqualTo(flowGoVersion, entry.FlowGoVersion) {
+			cadenceVersion = entry.CadenceVersion
+		} else {
+			break
```

Comment:
Why `break` here? 

It seems we always just look at the very first version entry only, in that case, why not just having one version entry instead of a mapping?

If we are going to hardcode a version map entry, I would say we just hard code the default one:

```
var DefaultVersionMapEntry = VersionMapEntry{ 
  FlowGoVersion: semver.Version{"0.0.0"}, CadenceVersion: semver.Version{"0.0.0"} }

// change this with versions FlowGoVersion: F, and CadenceVersion: C, where
// CadenceVersion C had a feature toggle which won't be enabled if FlowGoVersion is below F.
var RequiredVersionMapEntry = VersionMapEntry{ 
  FlowGoVersion: semver.Version{"0.0.0"}, CadenceVersion: semver.Version{"0.0.0"} }
```

then we can simplify this function into:
```
func mapToCadenceVersion(flowGoVersion semver.Version, versionMapping VersionMapEntry) semver.Version {
  if greateThanOrEqualTo(flowGoVersion, versionMapping.FlowGoVersion) {
    return versionMapping.CadenceVersion
  }
  return DefaultVersionEntry.CadenceVersion
}
```